### PR TITLE
`DoesIntersect` - add bbox check

### LINKED ISSUE #1435: DoesInteresect - improve performance
#### Description:

Improve performance of `DoesIntersect` method by adding boudning box check before solid check.

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #1435 

Add bbox check before solid check to improve performance.

### CODE REVIEW

File: Revit_Core_Engine/Query/DoesIntersect.cs [hunk 1]
```diff
@@ -82,30 +82,29 @@ public static bool DoesIntersect(this BoundingBoxXYZ bbox, Element element)
             }
 
             Transform transform = element.Document.LinkTransform();
+            Outline outline = new Outline(bbox.Min, bbox.Max);
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/DoesIntersect.cs:85]: Looks like you are missing the following lines here?

```C#
                if (bbox.Transform.IsTranslation)
                {
                    outline.MinimumPoint += bbox.Transform.Origin;
                    outline.MaximumPoint += bbox.Transform.Origin;
                }
```

I would actually clean up the code a bit:
```C#
Outline outline = new Outline(bbox.Min, bbox.Max);
if (bbox.Transform.IsTranslation)
{
    outline.MinimumPoint += bbox.Transform.Origin;
    outline.MaximumPoint += bbox.Transform.Origin;
}
else
{
    BH.Engine.Base.Compute.RecordWarning("Intersection of the bounding boxes could not be checked. Only translation and identity transformation is currently supported.");
    return false;
}
```

**Review by pawelbaran — CHANGES_REQUESTED** [review]

File: Revit_Core_Engine/Query/DoesIntersect.cs (hunk 1 above)
Comment (michal-pekacki) [reply Revit_Core_Engine/Query/DoesIntersect.cs]: I added missing part 👍

**Review by pawelbaran — APPROVED** [review]
Changes make sense now, happy to approve 👍

**Comment by pawelbaran** [comment]
@BHoMBot check compliance

**Comment by pawelbaran** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 20857261724

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**